projects/Mallard/src/services/exceptions.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * We want to handle 500s in resilient ways so we need to know
+ * if service errors are 500s and maybe try and read cached data
+ */
+
+export class Error5XX extends Error {
+    constructor() {
+        super(__DEV__ ? 'Internal server error' : 'Something went wrong')
+    }
+}
+
+export class Timeout extends Error {
+    constructor() {
+        super(__DEV__ ? 'Timeout' : 'Something went wrong')
+    }
+}
+
+export const withTimeout = <T>(promise: Promise<T>, timeout: number) =>
```

Comment:
I would have used an `AbortController` here and tailored this more to a `fetch` call, but React Native doesn't have it as far as I'm aware 😞